experimenter/experimenter/nimbus-ui/src/components/PageSummary/TableSignoff/index.tsx

```diff
@@ -1,24 +1,41 @@
-/* This Source Code Form is subject to the terms of the Mozilla Public
- * License, v. 2.0. If a copy of the MPL was not distributed with this
- * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
-
 import React from "react";
 import { Card, Table } from "react-bootstrap";
+import Form from "react-bootstrap/Form";
 import LinkExternal from "src/components/LinkExternal";
 import { EXTERNAL_URLS } from "src/lib/constants";
 import { getExperiment_experimentBySlug_signoffRecommendations } from "src/types/getExperiment";
 
 type TableSignoffProps = {
   signoffRecommendations: getExperiment_experimentBySlug_signoffRecommendations | null;
+  legalSignoff: boolean | null;
+  qaSignoff: boolean | null;
+  vpSignoff: boolean | null;
+  onLegalSignoffChange: (value: boolean) => void;
+  onQaSignoffChange: (value: boolean) => void;
+  onVpSignoffChange: (value: boolean) => void;
 };
 
-const TableSignoff = ({ signoffRecommendations }: TableSignoffProps) => (
+const TableSignoff = ({
+  signoffRecommendations,
+  legalSignoff,
+  qaSignoff,
+  vpSignoff,
+  onLegalSignoffChange,
+  onQaSignoffChange,
+  onVpSignoffChange,
+}: TableSignoffProps) => (
   <Card.Body>
     <Table data-testid="table-signoff">
       <tbody>
         <tr data-testid="table-signoff-qa">
           <th className="border-top-0">QA Sign-off</th>
-          <td className="border-top-0">
+          <td className="border-top-0 d-flex">
+            <Form.Check
+              type="checkbox"
+              data-testid="is-qasignoff-checkbox"
+              checked={!!qaSignoff}
+              onChange={(e) => onQaSignoffChange(e.target.checked)}
```

Comment:
this should be `onChange={onQaSignoffChange}` because this is creating a new fn every render, causing re-renders.

onQaSignoffChange should be changed to take an event.

Likewise for other handlers.

Reply:
Interesting! I've often used inline arrow functions just like this because it is more readable (to me at least), but if there's a significant performance impact that seems like a good reason to avoid. I can see how this would use a bit more memory to continually recreate the function, but curious how it causes re-renders, could you explain or do you have a link?

Reply:
When react renders a tree, for each child element it checks the old vs new props. If any props change (determined via a == comparision), then it detects the subtree has changed and requires re-rendering.

```
> x = () => {}
[Function: x]
> y = () => {}
[Function: y]
> x == y
false
```

Even though the functions are the same, they compare different, so it will always trigger a re-render.